src/besselk.jl

```diff
@@ -271,3 +271,52 @@ function besselk_ratio_knu_knup1(v, x::T) where T
     xinv = inv(x)
     return xinv * (v + x + 1/2) + xinv * hn
 end
+
+# originally contribued by @cgeoga https://github.com/cgeoga/BesselK.jl/blob/main/src/besk_ser.jl
+function _besselk_ser(v, x, maxit, tol, modify)
+  T    = promote_type(typeof(x), typeof(v))
+  out  = zero(T)
+  oneT = one(T)
+  twoT = oneT+oneT
+  # precompute a handful of things:
+  xd2      = x/twoT
+  xd22     = xd2*xd2
+  half     = oneT/twoT
+  if modify
+    e2v      = exp2(v)
+    xd2_v    = (x^(2*v))/e2v
+    xd2_nv   = e2v
+  else
+    lxd2     = log(xd2)
+    xd2_v    = exp(v*lxd2)
+    xd2_nv   = exp(-v*lxd2)
+  end
+  gam_v    = gamma(v)
+  gam_nv   = gamma(-v)
+  gam_1mv  = -gam_nv*v # == gamma(one(T)-v)
+  gam_1mnv = gam_v*v   # == gamma(one(T)+v)
+  xd2_pow  = oneT
+  fact_k   = oneT
+  floatk   = convert(T, 0)
+  (gpv, gmv) = (gam_1mnv, gam_1mv)
```

Comment:
this doesn't do anything?

Reply:
Maybe I'm misunderstanding the question, but those values are used inside the loop. I moved them closer to the loop declaration in the version posted here.

Reply:
`gpv` and `gmv` are used after this, but `gam_1mnv`  and `gam_1mv` aren't, so this is just a renaming.

Reply:
Removed this extra assignment